tests/tests_core/test_experiment.py

```diff
@@ -360,108 +360,86 @@ def test_outlier_filtering_derived_kpi(self):
-        self.assertEqual(result[3]['C'], 17)
-        self.assertEqual(result[3]['D'], 19)
-        self.assertEqual(result[3]['E'], 23)
+        self.assertAlmostEqual(result[1], 0.05357161207695437)
 
-        self.assertTrue(all([val == 20 for val in result[4]]))
-
-    def test_chi_square_test_result_and_statistics_different_weights(self):
+    def test_run_goodness_of_fit_test_false(self):
         exp = self.getExperiment()
-        data = ['A'] * 23 + ['B'] * 18 + ['C'] * 17 + ['D'] * 19 + ['E'] * 23
-        weights = {'A':  0.25, 'B':  0.15, 'C':  0.10, 'D':  0.40, 'E':  0.10}
-        result = exp.chi_square_test_result_and_statistics(data, weights)
+        observed_freqs = pd.Series([230, 180, 170, 190, 230], ['A', 'B', 'C', 'D', 'E'])
+        expected_freqs = pd.Series([250, 150, 100, 400, 100], ['A', 'B', 'C', 'D', 'E'])
+        result = exp.run_goodness_of_fit_test(observed_freqs, expected_freqs)
         self.assertEqual(result[0], False)
-        self.assertAlmostEqual(result[1], 9.064563321754584e-07)
-        self.assertAlmostEqual(result[2], 33.585)
-        self.assertEqual(len(result[3]), 5)
-        self.assertEqual(len(result[4]), 5)
+        self.assertAlmostEqual(result[1], 1.5123889771594655e-147)
 
-    def test_chi_square_test_result_and_statistics_2_categories(self):
+    def test_run_goodness_of_fit_test_2_variants(self):
         exp = self.getExperiment()
-        data = ['A'] * 17 + ['B'] * 17
-        weights = {'A': 0.5, 'B': 0.5}
-        result = exp.chi_square_test_result_and_statistics(data, weights)
+        observed_freqs = pd.Series([170, 170], ['A', 'B'])
+        expected_freqs = pd.Series([170, 170], ['A', 'B'])
+        result = exp.run_goodness_of_fit_test(observed_freqs, expected_freqs)
         self.assertEqual(result[0], True)
         self.assertAlmostEqual(result[1], 1.0)
-        self.assertAlmostEqual(result[2], 0.0)
-        self.assertEqual(result[3]['A'], 17)
-        self.assertEqual(result[4]['A'], 17)
-        self.assertEqual(result[3]['B'], 17)
-        self.assertEqual(result[4]['B'], 17)
-
-    def test_chi_square_test_result_and_statistics_NaN_data(self):
-        exp = self.getExperiment()
-        data = ['A'] * 17 + [np.nan] * 17
-        weights = {'A': 0.5, 'B': 0.5}
-        with self.assertRaises(ValueError):
-            exp.chi_square_test_result_and_statistics(data, weights)
 
-    def test_chi_square_test_result_and_statistics_counts_less_5_2_categories(self):
+    def test_run_goodness_of_fit_test_NaN_data(self):
         exp = self.getExperiment()
-        data = ['A'] * 17 + ['B'] * 2 + ['C'] * 3
-        weights = {'A': 0.33, 'B': 0.33, 'C': 0.33}
+        observed_freqs = pd.Series([170, 170], ['A', np.nan])
+        expected_freqs = pd.Series([170, 170], ['A', 'B'])
         with self.assertRaises(ValueError):
-            exp.chi_square_test_result_and_statistics(data, weights)
+            exp.run_goodness_of_fit_test(observed_freqs, expected_freqs)
```

Comment:
1. from which line will it raise a ValueError? Maybe you can also assert the error message so that we are sure it raises at the expected line.

2. Will it raise an error when:
```
observed_freqs = pd.Series([170, 170], ['A', 'C'])
expected_freqs = pd.Series([170, 170], ['A', 'B'])
```

Reply:
1. yes, will adapt for the incorrect input
2. yes, it will, because I always filter expected_freqs by observed_freqs keys.

Reply:
2. cool. could you also add a test to confirm it?